src/Compression-Tests/ZipArchiveTest.class.st

```diff
@@ -224,6 +224,8 @@ ZipArchiveTest >> testSetLastModification [
 { #category : #tests }
 ZipArchiveTest >> testShouldUnzipAndOverwriteWithoutInforming [
 	| nestedFileToZip fileModificationTime fileModificationTimeBlock |
+	Smalltalk os isWindows
+		ifTrue: [ "Occassionally trying to delete an existing file fails in Windows due to a file open lock" ^ self skip ].
```

Comment:
I don't think skipping the tests on windows is really a good solution because this should also run on windows... The test should be written in a manner that:
 - the tests should close all files open during the test
 - zip should close any file it opens (for reading or writing)

Reply:
Hi, I agree skipping the tests in windows is not great but I believe this is more to do with the Windows file primitives than to do with the current code.
Unfortunately I don't have access to a windows system so will just be more aggressive in closing any resources.

As far as I can see
 - The tests do close any files that are opened and deletes them.
 - The zip does close any files it opens.
Minimal changes were made to the file handling operations; the code was restructured to remove the ui parts. Lets see if the build goes through now...